src/main/kotlin/day18/data_source/PhotoDataSource.kt

```
@@ -0,0 +1,628 @@
+        val data = Json.decodeFromString<PixabayResponse>(textData)
+        return data.hits ?: throw NullPointerException("exception in PhotoDataSourceImpl.getPhotos()")
+    }
+}
+
+class MockPhotoDataSourceImpl : PhotoDataSource {
+    private val client = HttpClient(CIO)
+    override suspend fun getPhotos(query: String): List<PhotoDto> {
+
+        // 빈 검색어 예외 상황 처리
+        if (query.isEmpty()) {
+            throw IllegalArgumentException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)